passes/src/main/scala/at/forsyte/apalache/tla/passes/imp/SanyParserPassImpl.scala

```diff
@@ -89,25 +95,29 @@ class SanyParserPassImpl @Inject() (
     }
   }
 
+  override def dependencies = Set()
+
+  override def transformations = Set()
+
   override def execute(module: TlaModule): PassResult = {
-    import SourceOption._
-    val src = options.input.source
-    for {
-      rootModule <- src.format match {
-        case Format.Itf  => throw new SanyImporterException("Parsing the ITF format is not supported")
-        case Format.Json => loadFromJsonSource(src)
-        case Format.Tla =>
-          src match {
-            case StringSource(content, aux, _) => loadFromTlaString(content, aux)
-            case FileSource(file, _)           => loadFromTlaFile(file)
-          }
-      }
-      sortedModule <- sortDeclarations(rootModule)
-      _ <- saveLoadedModule(sortedModule)
-    } yield sortedModule
+    try {
+      parseSource(options.input.source)
+    } catch {
+      case err: SanyException =>
```

Comment:
Could you explain why we catch `SanyException` here, but not `AnnotationParserError` (like in `ParserExceptionAdapter`)?

Reply:
I just didn't realize that this would be needed, and no tests or static analysis was able to warn me. Thank you for catching this!

Reply:
😬